code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -18,19 +18,20 @@
 	//Placeholder for effect that trigger on eating that aren't tied to reagents.
 /obj/item/weapon/reagent_containers/food/snacks/proc/On_Consume(var/mob/M)
 	if(!usr)	return
-	if(!reagents.total_volume)
-		if(M == usr)
-			usr << "<span class='notice'>You finish eating \the [src].</span>"
-		usr.visible_message("<span class='notice'>[usr] finishes eating \the [src].</span>")
-		usr.unEquip(src)	//so icons update :[
-
-		if(trash)
-			if(ispath(trash,/obj/item))
-				var/obj/item/TrashItem = new trash(usr)
-				usr.put_in_hands(TrashItem)
-			else if(istype(trash,/obj/item))
-				usr.put_in_hands(trash)
-		qdel(src)
+	spawn(0)

```

Comment:
@Fox-McCloud It's to counteract the spawn(0) in the consume proc (which was in  the original code, and I wasn't sure about taking out). Without it the code below, including the qdel, runs before the reagents have been transferred